sympy/core/tests/test_function.py

```diff
@@ -175,12 +176,12 @@ def test_Lambda():
     assert e(y) == y**2
 
     assert Lambda((), 42)() == 42
-    assert Lambda((), 42) == Lambda((), 42)
+    assert unchanged(Lambda, (), 42)
     assert Lambda((), 42) != Lambda((), 43)
     assert Lambda((), f(x))() == f(x)
     assert Lambda((), 42).nargs == FiniteSet(0)
 
-    assert Lambda(x, x**2) == Lambda(x, x**2)
+    assert unchanged(Lambda, (x,), x**2)
```

Comment:
In hindsight, a test as `Lambda(x, x**2) == Lambda((x,), x**2)` could have been added. If I get to do any revisions, I'll add it next time (or in a later PR if I remember).